### DISCUSSION TITLE
Sketchpad base64 encoding doesn't fit in excel cells

### DISCUSSION THREAD

**Discussion by sshelat** [jspsych/jsPsych#3098] [Q&A]
Hi jsPsych folks,

I'm writing an experiment that involves users drawing with the sketchpad plugin, but I'm noticing that most drawings (i.e., drawings that have more colors and more content) don't fit in one cell when downloaded to csv. The base64 string is rather split over several lines, and it's making data-reading a bit more complex. See image below:

![image](https://github.com/jspsych/jsPsych/assets/52308720/9d520b62-d2f0-4ca6-a829-567b0c05b629)

The whole base64 string should be after "drawingSeen", but it breaks, skips a row, and writes it into the first cell of that new row. Does anything have any ideas on how I could fix this? Happy to provide more details. Pinging @jodeleeuw

**Comment by jodeleeuw** [comment]
Hi @sshelat,

Is this just an Excel visualization issue? If you hide the overflow in the cell it might fix it.
![Recording 2023-07-17 at 11 26 27](https://github.com/jspsych/jsPsych/assets/595524/994dbb35-062c-4c6a-a291-13f25a9d82da)

**Reply by sshelat** [reply]
Thanks for the reply @jodeleeuw. It's actually breaking the base64 encoding into different rows, not just spilling it over. See example for when I hide overflow in the first cell:

![image](https://github.com/jspsych/jsPsych/assets/52308720/bc9f3002-c707-4dc3-927c-820aea5d6903)

Then, when I select the first cell of the row underneath, you can see that it picks up where the base64 was cut off.

![image](https://github.com/jspsych/jsPsych/assets/52308720/dc28b815-99df-4abf-9b9d-eaeba4780734)

Any thoughts? I'm thinking I may have to just post-process the json outputs of my experiment to pull out my drawings (rather than the csv in .xlsx form).

**Reply by jodeleeuw** [reply]
Hmm, that's odd. Can you send me one of the raw CSV files from the experiment or post the first couple rows of data here? I'm wondering if this is an error in how jsPsych generated the CSV string or if it is a post-processing thing with really large strings.

Using JSON seems like the right approach to me. I find it to be the more reliable format for complicated datasets.

**Reply by sshelat** [reply]
Here's some sample data.

[sampledata.csv](https://github.com/jspsych/jsPsych/files/12073933/sampledata.csv)

**Reply by sshelat** [reply]
@jodeleeuw Have you found anything? I've written a postprocessing script that directly accesses the CSV and it seems to be working properly in converting the base64 to png. I think that suggests that this not a problem on jsPsych's end, but rather just a string cap in Excel for cells.
